### ISSUE TITLE
Get resolved color for current scheme in javascript land

### ISSUE THREAD

**Issue by danilobuerger** [klarna-incubator/platform-colors#20]
Motivation:

Some features of rn don't currently allow to use platform color: for example `tintColor` for `ActionSheetIOS` or interpolating colors for animations. For those use cases it would be nice to get the resolved color value for the current scheme (`Appearance.getColorScheme()`). 

Example:

```
// platform-colors.config.js
module.exports = {
  "colors": {
    "background": {
      "light": "#ffffff",
      "dark": "#000000"
    },
    "accent": "pink"
  },
}

// somewhereelse.js
resolveColor("background")
resolveColor(Colors.background)
```

In this case `resolveColor` would query `Appearance.getColorScheme()` for the appropriate scheme and then return the actual color value. 

Of course, this could already be done by importing the config and selecting from `colors` but it would be nice to have a importable function and type checked color names for this.

**Comment by oblador** [comment]
This would be a native module then? Do you think it makes sense for this functionality to be in `react-native` core or should it be a part of this library? (I know your feelings about contributing to core...)

**Comment by danilobuerger** [comment]
If we exclude the high contrast colors for now, this can already be done in js-land by just using `Appearance.getColorScheme()`. The way that I have currently set this up:

```ts
interface DynamicColor {
  dark: string;
  light: string;
}

export const dynamicColors = {
  background: { dark: "rgb(18,18,18)", light: "rgb(255,255,255)" },
}

export function resolveDynamicColor(color: DynamicColor): string {
  const scheme = Appearance.getColorScheme();
  return scheme === "dark" ? color.dark : color.light;
}
```
then it can be resolved by using:
```ts
resolveDynamicColor(dynamicColors.background);
```

Regarding high contrast, this could / should probably be added to `Appearance` in the core and can then be made available here.

**Comment by oblador** [comment]
Well, problem is that you then wouldn't have a single source of truth. It's quite possible for JS and native colors to get out of sync, so I think I would prefer a native module in that case.

**Comment by danilobuerger** [comment]
Isn't the single source of truth `platform-colors.config.js`? The rest is just generated code.

**Comment by oblador** [comment]
It's the source of truth when generating the files, but in runtime it's the generated file. Also a dynamic color might be manually maintained, or provided by the OS itself too

**Comment by oblador** [comment]
Closing as this is resolved in 0.2.0.
